pkg/tsdb/prometheus/resource/resource.go

```diff
@@ -34,14 +84,46 @@ func New(
 
 	p := client.NewProvider(settings, jsonData, httpClientProvider, cfg, features, plog)
 
+	customHeaders := make(map[string]string)
+	var jsonDataMap map[string]interface{}
+
+	err := json.Unmarshal(settings.JSONData, &jsonDataMap)
+	if err != nil {
+		return nil, err
+	}
+
+	index := 1
+	for {
```

Comment:
i have a bias against infinite loops. although the logic in this one seems solid now, it might be worth adding a test to help catch any issues if it's modified in the future.